[DOCS] Organize breaking changes by component (#79162)
elastic/elasticsearch: docs/reference/migration/migrate_8_0/_cluster-node-setting-changes.asciidoc

[discrete]
[[breaking_80_cluster_node_setting_changes]]
==== Cluster and node setting changes

//NOTE: The notable-breaking-changes tagged regions are re-used in the
//Installation and Upgrade Guide

//tag::notable-breaking-changes[]
TIP: {ess-setting-change}

> .You can no longer set `xpack.searchable.snapshot.shared_cache.size` on non-frozen nodes. {ess-icon}

Author: TODO: This is whitelisted as a safe Cloud setting, but I don't think it makes much sense. I need to follow up with the Distributed team before publishing.

Author: I reached out to the Distributed team. We can handle this fix as a separate work effort. The solution will likely impact Cloud users so I'll leave the icon for now.

[%collapsible]
====
*Details* +

> Setting `xpack.searchable.snapshot.shared_cache.size` to be positive on a node
> that does not have the `data_frozen` role was deprecated in {es} 7.12.0 and has
> been removed in {es} 8.0.0.

Reviewer: ```suggestion
Setting `xpack.searchable.snapshot.shared_cache.size` on a node
that does not have the `data_frozen` role was deprecated in {es} 7.12.0 and has
been removed in {es} 8.0.0.
```

*Impact* +
{es} only allocates partially mounted indices to nodes with the `data_frozen`

> role. Do not set `xpack.searchable.snapshot.shared_cache.size` on nodes without
> the `data_frozen` role. Removing the setting on nodes without the `data_frozen`
> role will not impact functionality.

Reviewer: I'd frame this as what they should do.

```suggestion
role. Remove the `xpack.searchable.snapshot.shared_cache.size` setting from nodes that don't have the `data_frozen` role. 
```

====

.`action.destructive_requires_name` now defaults to `false`. {ess-icon}
[%collapsible]
====
*Details* +

> The default value of the setting `action.destructive_requires_name` changes from `false`
> to `true` in {es} 8.0.0.
> 
> In previous versions, the default setting allowed users to use wildcard
> patterns to delete, close, or change index blocks on indices. In order
> to prevent the accidental deletion of indices that happen to match a
> wildcard pattern, we now require, by default, that any such destructive
> operation explicitly name the indices it intends to modify.

Reviewer: ```suggestion
The default for the `action.destructive_requires_name` setting changes from `false`
to `true` in {es} 8.0.0.

Previously, defaulting to `false` allowed users to use wildcard
patterns to delete, close, or change index blocks on indices. 
To prevent the accidental deletion of indices that happen to match a
wildcard pattern, we now default to requiring that destructive
operations explicitly name the indices to be modified.
```

*Impact* +

> If you would like to use wildcard patterns for destructive actions, set
> `action.destructive_requires_name` to `false` using the
> {ref}/cluster-update-settings.html[] cluster settings API].

Reviewer: ```suggestion
To use wildcard patterns for destructive actions, set
`action.destructive_requires_name` to `false` using the
{ref}/cluster-update-settings.html[] cluster settings API].
```

====

[[max_clause_count_change]]
.The `indices.query.bool.max_clause_count` setting now limits all query clauses.
[%collapsible]
====
*Details* +
Previously, the `indices.query.bool.max_clause_count` would apply to the number
of clauses of a single `bool` query. It now applies to the total number of

> clauses of the rewritten query. In order to reduce chances of breaks, its
> default value has been bumped from 1024 to 4096.
> 
> *Impact* +
> Queries with many clauses should be avoided whenever possible. If you had bumped
> this setting already in order to accomodate for some heavy queries, you might
> need to bump it further so that these heavy queries keep working.

Reviewer: ```suggestion
clauses of the rewritten query. To reduce chances of breaks, its
default value has been bumped from 1024 to 4096.

*Impact* +
Queries with many clauses should be avoided whenever possible. 
If you previously bumped this setting to accommodate heavy queries, 
you might need to increase it further. 
```

====

[[ilm-poll-interval-limit]]
.`indices.lifecycle.poll_interval` must be greater than `1s`.
[%collapsible]
====
*Details* +

> The setting `indices.lifecycle.poll_interval`, if set too low, can cause
> excessive load on a cluster. This setting must now be set to `1s` (one second)
> or greater.

Reviewer: ```suggestion
Setting `indices.lifecycle.poll_interval` too low can cause
excessive load on a cluster. The poll interval must now be at least `1s` (one second).
```

*Impact* +

> Update the `indices.lifecycle.poll_interval` setting to a value of `1s` or
> greater using `elasticsearch.yml` or the
> {ref}/cluster-update-settings.html[cluster update settings API].

Reviewer: ```suggestion
Set `indices.lifecycle.poll_interval` setting to `1s` or
greater in `elasticsearch.yml` or through the
{ref}/cluster-update-settings.html[cluster update settings API].
```

[source,yaml]
----
xpack.security.authc.realms:

  native.realm1.enabled: false
  file.realm2.enabled: false

  ...
----
====

.The realm `order` setting is now required.
[%collapsible]
====
*Details* +
The `xpack.security.authc.realms.{type}.{name}.order` setting is now required and must be
specified for each explicitly configured realm. Each value must be unique.

*Impact* +
The cluster will fail to start if the requirements are not met.

For example, the following configuration is invalid:
[source,yaml]
--------------------------------------------------
xpack.security.authc.realms.kerberos.kerb1:
  keytab.path: es.keytab
  remove_realm_name: false
--------------------------------------------------

And must be configured as:
[source,yaml]
--------------------------------------------------
xpack.security.authc.realms.kerberos.kerb1:
  order: 0
  keytab.path: es.keytab
  remove_realm_name: false
--------------------------------------------------
====

[[breaking_80_allocation_change_include_relocations_removed]]
.`cluster.routing.allocation.disk.include_relocations` has been removed.
[%collapsible]
====
*Details* +

> By default {es} will account for the sizes of relocating shards when making
> allocation decisions based on the disk usage of the nodes in the cluster. In
> earlier versions the `cluster.routing.allocation.disk.include_relocations`
> setting allowed this accounting to be disabled, which would result in poor
> allocation decisions that might overshoot watermarks and require significant
> extra work to correct. This behaviour is no longer optional, and this setting
> has been removed.

Reviewer: ```suggestion
{es} now always accounts for the sizes of relocating shards when making
allocation decisions based on the disk usage of the nodes in the cluster. In
earlier versions, you could disable this by setting `cluster.routing.allocation.disk.include_relocations` to `false`.
That could result in poor allocation decisions that could overshoot watermarks and require significant
extra work to correct. The `cluster.routing.allocation.disk.include_relocations` setting has been removed.
```

*Impact* +

> Discontinue use of the `cluster.routing.allocation.disk.include_relocations`
> setting. Specifying this setting in `elasticsearch.yml` will result in an error
> on startup.

Reviewer: ```suggestion
Remove the `cluster.routing.allocation.disk.include_relocations`
setting. Specifying this setting in `elasticsearch.yml` will result in an error
on startup.
```

====

.cluster.join.timeout` has been removed.
[%collapsible]
====
*Details* +
The `cluster.join.timeout` setting has been removed. Join attempts no longer
time out.

*Impact* +

> Do not set `cluster.join.timeout` in your `elasticsearch.yml` file.

Reviewer: ```suggestion
Remove `cluster.join.timeout` from `elasticsearch.yml`.
```

====

.`discovery.zen` settings have been removed.
[%collapsible]
====
*Details* +

> All settings under the `discovery.zen` namespace, which existed only for BWC reasons in 7.x,
> are no longer supported. In particular, this includes:
> 

Reviewer: ```suggestion
All settings under the `discovery.zen` namespace are no longer supported. They existed only only for BWC reasons in 7.x. This includes:

```

- `discovery.zen.minimum_master_nodes`
- `discovery.zen.no_master_block`
- `discovery.zen.hosts_provider`
- `discovery.zen.publish_timeout`
- `discovery.zen.commit_timeout`
- `discovery.zen.publish_diff.enable`
- `discovery.zen.ping.unicast.concurrent_connects`
- `discovery.zen.ping.unicast.hosts.resolve_timeout`
- `discovery.zen.ping.unicast.hosts`
- `discovery.zen.ping_timeout`
- `discovery.zen.unsafe_rolling_upgrades_enabled`
- `discovery.zen.fd.connect_on_network_disconnect`
- `discovery.zen.fd.ping_interval`
- `discovery.zen.fd.ping_timeout`
- `discovery.zen.fd.ping_retries`
- `discovery.zen.fd.register_connection_listener`
- `discovery.zen.join_retry_attempts`
- `discovery.zen.join_retry_delay`
- `discovery.zen.join_timeout`
- `discovery.zen.max_pings_from_another_master`
- `discovery.zen.send_leave_request`
- `discovery.zen.master_election.wait_for_joins_timeout`
- `discovery.zen.master_election.ignore_non_master_pings`
- `discovery.zen.publish.max_pending_cluster_states`

*Impact* +

> Discontinue use of the `discovery.zen` settings. Specifying these settings in
> `elasticsearch.yml` will result in an error on startup.

Reviewer: ```suggestion
Remove the `discovery.zen` settings from `elasticsearch.yml`. Specifying these settings will result in an error on startup.
```

====

.`http.content_type.required` has been removed.
[%collapsible]
====
*Details* +
The `http.content_type.required` setting was deprecated in Elasticsearch 6.0
and has been removed in Elasticsearch 8.0. The setting was introduced in
Elasticsearch 5.3 to prepare users for Elasticsearch 6.0, where content type
auto detection was removed for HTTP requests.

*Impact* +

> Discontinue use of the `http.content_type.required` setting. Specifying this
> setting in `elasticsearch.yml` will result in an error on startup.

Reviewer: ```suggestion
Remove the `http.content_type.required` setting from `elasticsearch.yml`. Specifying this setting  will result in an error on startup.
```

====

.`http.tcp_no_delay` has been removed.
[%collapsible]
====
*Details* +

> The `http.tcp_no_delay` setting was deprecated in 7.x and has been removed in 8.0. It has been replaced by
> `http.tcp.no_delay`.

Reviewer: ```suggestion
The `http.tcp_no_delay` setting was deprecated in 7.x and has been removed in 8.0. Use`http.tcp.no_delay` instead.
```

*Impact* +

> Use the `http.tcp.no_delay` setting. Discontinue use of the `http.tcp_no_delay`
> setting. Specifying the `http.tcp_no_delay` setting in `elasticsearch.yml` will
> result in an error on startup.

Reviewer: ```suggestion
Replace the `http.tcp_no_delay` setting with `http.tcp.no_delay`.  
Specifying  `http.tcp_no_delay` in `elasticsearch.yml` will
result in an error on startup.
```

====

.`network.tcp.connect_timeout` has been removed.
[%collapsible]
====
*Details* +
The `network.tcp.connect_timeout` setting was deprecated in 7.x and has been removed in 8.0. This setting
was a fallback setting for `transport.connect_timeout`.

*Impact* +

> Use the `transport.connect_timeout` setting to change the default connection
> timeout for client connections. Discontinue use of the
> `network.tcp.connect_timeout` setting. Specifying the
> `network.tcp.connect_timeout` setting in `elasticsearch.yml` will result in an
> error on startup.

Reviewer: ```suggestion
Remove the`network.tcp.connect_timeout` setting. 
Use the `transport.connect_timeout` setting to change the default connection
timeout for client connections. Specifying
`network.tcp.connect_timeout` in `elasticsearch.yml` will result in an
error on startup.
```

====

.`node.max_local_storage_nodes` has been removed.
[%collapsible]
====
*Details* +
The `node.max_local_storage_nodes` setting was deprecated in 7.x and
has been removed in 8.0. Nodes should be run on separate data paths
to ensure that each node is consistently assigned to the same data path.

*Impact* +

> Discontinue use of the `node.max_local_storage_nodes` setting. Specifying this
> setting in `elasticsearch.yml` will result in an error on startup.

Reviewer: ```suggestion
Remove the `node.max_local_storage_nodes` setting. Specifying this
setting in `elasticsearch.yml` will result in an error on startup.
```

====

[[accept-default-password-removed]]
.The `accept_default_password` setting has been removed.
[%collapsible]
====
*Details* +

> The `xpack.security.authc.accept_default_password` setting has not had any affect
> since the 6.0 release of {es}. It has been removed and cannot be used.

Reviewer: ```suggestion
The `xpack.security.authc.accept_default_password` setting has not had any affect
since the 6.0 release of {es} and is no longer allowed.
```

*Impact* +

> Discontinue use of the `xpack.security.authc.accept_default_password` setting.
> Specifying this setting in `elasticsearch.yml` will result in an error on
> startup.

Reviewer: ```suggestion
Remove  the `xpack.security.authc.accept_default_password` setting from `elasticsearch.yml`.
Specifying this setting will result in an error on startup.
```

====

[[roles-index-cache-removed]]
.The `roles.index.cache.*` settings have been removed.
[%collapsible]
====
*Details* +
The `xpack.security.authz.store.roles.index.cache.max_size` and
`xpack.security.authz.store.roles.index.cache.ttl` settings have
been removed. These settings have been redundant and deprecated
since the 5.2 release of {es}.

*Impact* +

> Discontinue use of the `xpack.security.authz.store.roles.index.cache.max_size`
> and `xpack.security.authz.store.roles.index.cache.ttl` settings. Specifying
> these settings in `elasticsearch.yml` will result in an error on startup.

Reviewer: ```suggestion
Remove the `xpack.security.authz.store.roles.index.cache.max_size`
and `xpack.security.authz.store.roles.index.cache.ttl` settings from `elasticsearch.yml` . 
Specifying these settings will result in an error on startup.
```

====

[[separating-node-and-client-traffic]]
.The `transport.profiles.*.xpack.security.type` setting has been removed.
[%collapsible]
====
*Details* +

> The `transport.profiles.*.xpack.security.type` setting has been removed since
> the Transport Client has been removed and therefore all client traffic now uses
> the HTTP transport. Transport profiles using this setting should be removed.

Reviewer: ```suggestion
The `transport.profiles.*.xpack.security.type` setting is no longer supported.
The Transport Client has been removed and all client traffic now uses
the HTTP transport. Transport profiles using this setting should be removed.
```

*Impact* +

> Discontinue use of the `transport.profiles.*.xpack.security.type` setting.
> Specifying this setting in a transport profile in `elasticsearch.yml` will
> result in an error on startup.

Reviewer: ```suggestion
Remove the `transport.profiles.*.xpack.security.type` setting from `elasticsearch.yml`.
Specifying this setting in a transport profile will result in an error on startup.
```
